batched/dense/impl/KokkosBatched_Tbsv_Serial_Impl.hpp

```diff
@@ -0,0 +1,170 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#ifndef KOKKOSBATCHED_TBSV_SERIAL_IMPL_HPP_
+#define KOKKOSBATCHED_TBSV_SERIAL_IMPL_HPP_
+
+/// \author Yuuichi Asahi (yuuichi.asahi@cea.fr)
+
+#include "KokkosBatched_Util.hpp"
+#include "KokkosBatched_Tbsv_Serial_Internal.hpp"
+
+namespace KokkosBatched {
+
+template <typename AViewType, typename XViewType>
+KOKKOS_INLINE_FUNCTION static int checkTbsvInput(
+    [[maybe_unused]] const AViewType &A, [[maybe_unused]] const XViewType &x,
+    [[maybe_unused]] const int k, [[maybe_unused]] const int incx) {
+#if (KOKKOSKERNELS_DEBUG_LEVEL > 0)
+  static_assert(Kokkos::is_view<AViewType>::value,
```

Comment:
Static assert are a compile time feature, why do you have them protected behind `KOKKOSKERNELS_DEBUG_LEVEL`? These compile time features do not increase the execution of the compiler code, I would move the check out of the pre-processor guard

Reply:
I fully agree with you. I will fix accordingly. I found static assertions are protected in [some function](https://github.com/kokkos/kokkos-kernels/blob/d1a91b8a1f3aa4a972d7ff198bce73ec3248f641/batched/dense/impl/KokkosBatched_HadamardProduct_Impl.hpp#L94-L109), so I implement this way for consistency.